pycloudlib/gce/cloud.py

```diff
@@ -47,8 +48,22 @@ def __init__(
         self.compute = googleapiclient.discovery.build(
             'compute', 'v1', cache_discovery=False
         )
-        if not all([project, region, zone]):
-            raise ValueError('project, region, and zone are required!')
+        if not all([region, zone]):
+            raise ValueError('region, and zone are required!')
+        if not project:
+            command = ['gcloud', 'config', 'get-value', 'project']
```

Comment:
Another approach we could follow is to read the `project` directly from the json provided by `GOOGLE_APPLICATION_CREDENTIALS`. In that way, we would not need to setup up `gcloud`.